dhall-docs/src/Dhall/Docs/Html.hs

```diff
@@ -38,42 +57,115 @@ removeComments (Header header)
 filePathHeaderToHtml
     :: (Path Abs File, Header) -- ^ (source file name, parsed header)
-    -> FilePath           -- ^ Relative path to the css of the tool
+    -> DocParams               -- ^ Parameters for the documentation
     -> Html ()
-filePathHeaderToHtml (filePath, header) cssFile =
-    html_ $ do
-        head_ $ do
-            title_ $ toHtml title
-            stylesheet cssFile
+filePathHeaderToHtml (filePath, header) params@DocParams{..} =
+    doctypehtml_ $ do
+        headContents title params
         body_ $ do
-            h1_ $ toHtml title
-            headerToHtml header
+            navBar params
+            mainContainer $ do
+                h1_ $ toHtml title
+                headerToHtml header
   where
     title = Path.fromRelFile $ Path.filename filePath
 
 
 -- | Generates an index @`Html` ()@ that list all the dhall files in that folder
 indexToHtml
-    :: FilePath   -- ^ Index directory
-    -> [FilePath] -- ^ Generated files in that directory
-    -> FilePath   -- ^ Relative path to the css of the tool
+    :: FilePath        -- ^ Index directory
+    -> [Path Rel File] -- ^ Generated files in that directory
+    -> [Path Rel Dir]  -- ^ Generated directories in that directory
+    -> DocParams       -- ^ Parameters for the documentation
     -> Html ()
-indexToHtml dir files cssFile = html_ $ do
-    head_ $ do
-        title_ $ toHtml title
-        stylesheet cssFile
+indexToHtml indexDir files dirs params@DocParams{..} = doctypehtml_ $ do
+    headContents title params
     body_ $ do
-        h1_ $ toHtml title
-        p_ "Exported files: "
-        ul_ $ mconcat $ map (li_ . toHtml) files
+        navBar params
+        mainContainer $ do
+            h1_ $ toHtml title
+
+            Control.Monad.unless (null files) $ do
+                h2_ "Exported files: "
+                ul_ $ mconcat $ map listFile files
+
+            Control.Monad.unless (null dirs) $ do
+                h2_ "Exported packages: "
+                ul_ $ mconcat $ map listDir dirs
 
   where
+    listFile :: Path Rel File -> Html ()
+    listFile file =
+        let fileRef = Data.Text.pack $ Path.fromRelFile file
+            itemText = Data.Text.pack $ tryToTakeExt file
+        in li_ $ a_ [href_ fileRef] $ toHtml itemText
+
+    listDir :: Path Rel Dir -> Html ()
+    listDir dir =
+        let filePath = Data.Text.pack $ Path.fromRelDir dir in
+        li_ $ a_ [href_ (filePath <> "index.html")] $ toHtml filePath
+
+    tryToTakeExt :: Path Rel File -> FilePath
+
+#if MIN_VERSION_path(0,7,0)
+    tryToTakeExt file = Path.fromRelFile $ case Path.splitExtension file of
+        Nothing -> file
+        Just (f, _) -> f
+#else
+    -- Sadly `path` < 0.7.0 doesn't provide a way to remove the extension
+    -- so we have to convert it to `FilePath` to do further manipulations
+    tryToTakeExt file = FilePath.replaceExtension (Path.fromRelFile file) ""
```

Comment:
Do we need to support path <0.7?

Reply:
Yes because path >= 0.7 requires some dependencies that are not compatible with ghc 8.2.0 I think. Unless we don't want dhall-docs to be compatible with it.

Reply:
If this would simplify things, we could consider removing support for GHC 8.2 BTW.

Reply:
ok let me think about it. That way I don't have to add a lot of boilerplate

Reply:
I dropped support for GHC 8.2. I'm pretty sure I'll make heavy use of this package and I don't want to keep doing this workarounds every time.